src/main/java/com/igot/cb/masterdata/controller/MasterDataController.java

```diff
@@ -44,4 +48,37 @@ public ResponseEntity<ApiResponse> updateDegree(
         return new ResponseEntity<>(masterDataService.updateDegreesList(authToken, requestBody), HttpStatus.OK);
     }
 
+    @PostMapping(value = "/degree/search")
+    public ResponseEntity<ApiResponse> searchDegree(@RequestBody SearchCriteria searchCriteria) {
+        return new ResponseEntity<>(masterDataService.searchDegree(searchCriteria), HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/institute/search")
+    public ResponseEntity<ApiResponse> searchInstitute(@RequestBody SearchCriteria searchCriteria) {
+        return new ResponseEntity<>(masterDataService.searchInstitute(searchCriteria), HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/add/degree")
+    public ResponseEntity<ApiResponse> addDegree(@RequestBody Degree degree) {
```

Comment:
why are you not following the default sunbird approach... using Map as RequestBody with format like,

{
"request": {
//necessary attributes...
}
}

Do not introduce model objects -- as this will reduce the code coverage lines or you need to add the test cases for this model class alone again which is not required...